uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/instances/leave-request.instance.js

```diff
@@ -11,6 +11,71 @@ define([
   instances.factory('LeaveRequestInstance', ['$q', 'checkPermissions', 'OptionGroup',
     'shared-settings', 'ModelInstance', 'LeaveRequestAPI',
     function ($q, checkPermissions, OptionGroup, sharedSettings, ModelInstance, LeaveRequestAPI) {
+      /**
+       * Amends balance change breakdown if calculation unit absence type is "hours".
+       * It changes the balance first and last days (only first day in case of single day request)
+       *   by setting the selected deductions.
+       * @NOTE this function mutates the "balanceChange" object
+       *
+       * @param  {Object} balanceChange
+       * @param  {String} calculationUnit
+       * @return {Promise} resolve with amended balance change in case of "hours" absence type
+       *   or returns unamended balance change in case of "days" absence type
+       */
+      function amendHourlyBalanceChangeBreakdown (balanceChange, calculationUnit) {
+        var breakdown = balanceChange.breakdown;
+        var request = this;
+
+        if (calculationUnit !== 'hours') {
```

Comment:
this check should not be part of the function, otherwise the name would be something like `amendBalanceChangeBreakdownIfUnitIsHourly` and obviously that would be a hint that the function does too many things.

Move the check out of the function

```js
if (calculationUnit === 'hours') {
  amendHourlyBalanceChangeBreakdown()l
}
```

Reply:
done.